torch/quantization/fx/pattern_utils.py

```diff
@@ -37,6 +37,18 @@ def get_default_quant_patterns():
 def get_default_output_activation_post_process_map():
     return DEFAULT_OUTPUT_ACTIVATION_POST_PROCESS_MAP
 
+# a set of QuantizeHandler classes that are not observed
+# we'll skip inserting observers for input and output for these QuantizeHandlers
+# used for ops that only supports dynamic/weight only quantization
+DEFAULT_NOT_OBSERVED_QUANTIZE_HANDLER = set()
+def mark_input_output_not_observed():
+    def insert(fn):
+        DEFAULT_NOT_OBSERVED_QUANTIZE_HANDLER.add(fn)
+        return fn
+    return insert
```

Comment:
question (not a suggestion that we do that, more asking for the sake of discussion): what are people's thoughts about defining this in terms of the implementation (current code), versus the use case (a mapping of ops to the quantization modes supported by those ops)?

Reply:
current code is trying to mark the ops that only works for dynamic/weight_only quant to bypass the insert observer for these ops, so we don't really need to complete list. Is the suggestions to add a complete specification for all ops? what is the motivation for that? If this is trying for documentation purpose I think we probably need to list them explicitly, here we are talking about QuantizeHandler, which is an implementation detail of fx graph mode quantization, not talking about specific module/ops actually.

Reply:
my question is about the conceptual level of how the mapping is defined, where the choices are (a) defined by concepts internal to the implementation, versus (b) defined by concepts used by the end users (which op supports which quantization type).  Currently this code is doing (a).  A risk with (a) is it requires an understanding of the implementation to modify it, so it will be harder for people who don't work on FX graph mode full time to make changes correctly.  If defined in terms of (b), that risk would be reduced.

In terms of implementation of (b), there are ways to do it with allowlist (specify all mappings) or blocklists (specify all the exceptions), I agree that using blocklists might be nicer.

It doesn't have to be in this PR, but something to consider for the future.

Reply:
Sure. I don't expect these cases to come up very often actually, we can do that if that becomes common in the future. Although I don't really feel that would help too much, since in that case we also need to define another mapping from ops/modules to QuantizeHandler.